near/src/runtime.rs

```diff
@@ -78,6 +78,21 @@ impl NightshadeRuntime {
     fn height_to_epoch(&self, height: BlockIndex) -> (Epoch, BlockIndex) {
         (height / self.genesis_config.epoch_length, height % self.genesis_config.epoch_length)
     }
+
+    fn get_block_proposer_info(&self, height: BlockIndex) -> Result<ValidatorStake, Box<std::error::Error>> {
+        let (epoch, idx) = self.height_to_epoch(height);
+        let mut vm = self.validator_manager.write().expect(POISONED_LOCK_ERR);
+        let validator_assignemnt = vm.get_validators(epoch)?;
+        let total_seats: u64 = validator_assignemnt.block_producers.iter().sum();
+        let mut cur_seats = 0;
+        for (i, seats) in validator_assignemnt.block_producers.iter().enumerate() {
+            if cur_seats + seats > idx % total_seats {
+                return Ok(validator_assignemnt.validators[i].clone());
+            }
+            cur_seats += seats;
+        }
+        unreachable!()
```

Comment:
Shouldn't this be some error instead of `unreachable!()`?

Reply:
That's the error - it should never reach it otherwise there is a bug in math.

Reply:
Okay. If that's the case why do you need to return `Result`